source/docs/advanced/graphql/slim-down-resolvers-with-loaders.md

```diff
@@ -0,0 +1,351 @@
+};
+```
+
+### Using GraphQL context for Dependency Injection
+
+In its current state, resolvers are tightly coupled to the current
+implementation. We recommend to leverage GraphQL’s context to inject a loader to
+resolvers.
+
+The GraphQL context is (according to
+[GraphQL.org](https://graphql.org/learn/execution/#root-fields-resolvers)):
+
+> A value which is provided to every resolver and holds important contextual
+> information like the currently logged in user, or access to a database.
+
+In Front-Commerce (and most GraphQL implementations), the `context` is passed as
+the 3rd argument of a resolver function.
+
+Let’s refactor resolvers to use a loader from the context instead of importing
+it:
+
+```diff
+// my-module/server/modules/clicks-counters/resolvers.js
+-import CounterLoader from "./loader";
+-
+module.exports = {
+  Product: {
+-    clicksCounter: ({ sku }) => CounterLoader.loadBySku(sku)
++    clicksCounter: ({ sku }, _, { loaders }) => {
++      return loaders.Counter.loadBySku(sku);
++    }
+  },
+
+  Mutation: {
+-    incrementProductCounter(_, { sku, incrementValue = 1 }) {
++    incrementProductCounter(_, { sku, incrementValue = 1 }, { loaders }) {
+-      return CounterLoader.incrementBySku(sku, incrementValue)
++      return loaders.Counter.incrementBySku(sku, incrementValue)
+        .then(() => ({
+          success: true
+        }));
+    }
+  }
+};
+```
+
+**You might be wondering why we used the `loaders` value from the context, and
+(more important) how this object was created… this is what we will learn in the
+next section!**
+
+### Registering a loader in Front-Commerce’s GraphQL context
+
+For the previous refactoring to work, we must register our loader in GraphQL’s
+context so the `loaders.Counter` could be used.
+
+In Front-Commerce’s GraphQL modules, you can achieve this using the
+[`contextEnhancer`](/docs/reference/graphql-module-definition.html#contextEnhancer-optional)
+key of your module definition.
+
+This function is called by Front-Commerce during each request for all GraphQL
+modules, to construct the GraphQL `context.loaders` object. Each module can
+build loaders and make them available in the context by returning an object to
+be merged with previously declared loaders.
+
+Let’s update the module definition to register a `Counter` loader:
+
+```diff
+// my-module/server/modules/clicks-counters/index.js
+import typeDefs from "./schema.gql";
+import resolvers from "./resolvers";
++import CounterLoader from "./loader";
+
+export default {
+  namespace: "ClicksCounters",
+  typeDefs: typeDefs
+  typeDefs: typeDefs,
+-  resolvers: resolvers
++  resolvers: resolvers,
++  contextEnhancer: () => {
++    return {
++      Counter: CounterLoader
++    };
++  }
+};
+```
+
+<blockquote class="info">
+  Please refer to the [GraphQL module definition
+  reference](/docs/reference/graphql-module-definition.html#contextEnhancer-optional)
+  for the exact signature and a more complex example.
+</blockquote>
```

Comment:
I had an "A-HA!" moment when I understood that loaders could use existing loaders.
It might be a good idea to mention this in a blockquote and to talk about dependencies.

edit: reading further, it might make sense to add it in the WIP section

Reply:
@JulienPradet sorry, I did not get it… could you please do it or explain your thoughts further?